### PR TITLE
Users list pagination

### LINKED ISSUE #12: Improve script which initializes bot with GitHub->Slack name mappings for active users
## Description

The `lib/retrieve-slack-github-users.js` script retrieves all the Slack users and fetches the respective profile to retrieve the GitHub ID. Currently the initial list fetch operation is not paginated, which poses a risk once Slack will clamp down on non-paginated API calls. The goal is to adapt the script so that it paginates the call using cursors.

## Implementation

The Slack API has a https://api.slack.com/methods/users.list method which will return all the members of the status-im Slack organization. That will return a complete list of users (which currently stands at around 2MB, so we should [use pagination](https://api.slack.com/docs/pagination#cursors)).

### PR DESCRIPTION
**Pull request by realphi**
fixes #12 

Added pagination to users list to avoid limits

### CODE REVIEW

**Comment by realphi** [comment]
fixes #12

File: lib/slack-profile-cache.js [hunk 1]
```diff
@@ -251,3 +252,16 @@ function timeout (ms) {
 async function sleep (timeoutMs) {
   await timeout(timeoutMs)
 }
+
+async function getUsersList (users, cursor) {
+  return slackWebClient.users.list({limit: 200, cursor: cursor}).then(
```
Comment (PombeirP) [inline lib/slack-profile-cache.js]: I see that the `getUsersList` function is already marked `async` - why not replace `then`/`catch` with `try`/`await`/`catch`?

Comment (realphi) [reply lib/slack-profile-cache.js]: ok, I'll do it!

File: lib/slack-profile-cache.js [hunk 2]
```diff
@@ -113,8 +114,8 @@ async function internalBuild (robot, cache) {
 
   try {
     const slackWeb = new WebClient(token) // We need to use a different token because users.profile API is not available to bot users
-    const usersList = await slackWeb.users.list() // TODO: This call should be paginated to avoid hitting limits (memory, API): https://api.slack.com/docs/pagination#cursors
-    const activeUsersList = usersList.members.filter(u => !u.deleted && !u.is_bot && u.id !== 'USLACKBOT')
+    const usersList = await getUsersList() // TODO: This call should be paginated to avoid hitting limits (memory, API): https://api.slack.com/docs/pagination#cursors
```
Comment (PombeirP) [inline lib/slack-profile-cache.js]: This TODO should be removed in this PR

File: lib/slack-profile-cache.js [hunk 3]
```diff
@@ -251,3 +252,18 @@ function timeout (ms) {
...
+    const res = await slackWebClient.users.list({limit: 200, cursor: cursor})
+    users = users.concat(res.members)
+
+    if (!res.response_metadata.next_cursor) {
+      return users
+    }
+
+    return getUsersList(users, res.response_metadata.next_cursor)
+  } catch (e) {
+    console.log(e)
```
Comment (PombeirP) [inline lib/slack-profile-cache.js]: We should rethrow the exception here (the top-level catch clause will already use  `robot.log.error` to log the message).

Comment (realphi) [reply lib/slack-profile-cache.js]: Oh I see, I didn't read the code thoroughly.

File: lib/slack-profile-cache.js [hunk 4]
```diff
@@ -251,3 +252,18 @@ function timeout (ms) {
...
+  try {
+    const res = await slackWebClient.users.list({limit: 200, cursor: cursor})
+    users = users.concat(res.members)
+
+    if (!res.response_metadata.next_cursor) {
+      return users
+    }
+
+    return getUsersList(users, res.response_metadata.next_cursor)
+  } catch (e) {
```
Comment (PombeirP) [inline lib/slack-profile-cache.js]: In this case, there's no point in catching the exception, we can just get rid of the `try/catch`

**Comment by realphi** [comment]
OK, I think it's finally done! 😀

**Comment by PombeirP** [comment]
Merged! Thanks for your contribution @realphi!

**Comment by realphi** [comment]
Thank you for guiding me, it was my first PR. I'll do more if I could resolve other issues.

**Pull request merged**